base/compiler/ssair/inlining.jl

```diff
@@ -875,8 +893,26 @@ function validate_sparams(sparams::SimpleVector)
     return true
 end
 
+function may_have_fcalls(m::Method)
+    may_have_fcall = true
+    if isdefined(m, :source)
+        src = m.source
+        isa(src, Vector{UInt8}) && (src = uncompressed_ir(m))
+        if isa(src, CodeInfo)
+            may_have_fcall = src.has_fcall
+        end
+    end
+    return may_have_fcall
+end
+
+function can_inline_typevars(m::MethodMatch, argtypes::Vector{Any})
+    may_have_fcalls(m.method) && return false
+    any(x -> x === Type && isdefined(x, :var), argtypes[2:end]) && return false
```

Comment:
```suggestion
    any(@nospecialize(x) -> x === Type && isdefined(x, :var), argtypes[2:end]) && return false
```
also, `x === Type && isdefined(x, :var)` can be simplified to `x === Type`.

Reply:
I guess you want `x isa UnionAll` instead?